### PR TITLE
Relative project path

### PR DESCRIPTION
**Pull request by techee**
OK, this is the last of my 1.5-year-old patches. In addition to absolute path this patch also saves relative path for every open file in the project. When loading, it first reads the absolute paths and if the file isn't found (e.g. the whole project directory was moved), it reads the relative path and tries to open the file relatively to the project file.

### CODE REVIEW

**Comment by frlan** [comment]
Just back on that... merge or close? ;)

**Comment by elextr** [comment]
On re-reading its hard to follow, but as far as I can tell, b4n made a suggestion to break the deadlock, but it hasn't been done, so no to merge.

**Comment by techee** [comment]
Closing. The way I'd like to go is https://github.com/geany/geany/pull/450 (once I have time to continue working on it) which makes this patch obsolete.
